tests/model_inheritance/tests.py

```diff
@@ -158,6 +161,40 @@ def test_create_diamond_mti_default_pk(self):
         with self.assertNumQueries(4):
             common_child.save()
 
+    def test_create_diamond_mti_common_parents(self):
+        with self.assertNumQueries(4):
+            itrmp = ItalianRestaurantManyParents.objects.create(
+                name="Ristorante Miron",
+                address="1234 W. Ash",
+            )
+
+        italian_resturant_place = itrmp.italianrestaurant_ptr.place_ptr
+        itrmp_place = itrmp.place_ptr_two
+        italian_resturant_resturant = itrmp.italianrestaurant_ptr.restaurant_ptr
+        itrmp_resturant = itrmp.restaurant_ptr
+
+        # Some draft tests to test regression
+        self.assertEqual(italian_resturant_place, itrmp_place)
+        self.assertEqual(italian_resturant_resturant, itrmp_resturant)
+        self.assertEqual(itrmp_resturant.place_ptr, itrmp_place)
+        self.assertEqual(itrmp.name, "Ristorante Miron")
+        self.assertEqual(itrmp.address, "1234 W. Ash")
+
+    def test_check_on_diamond_mti_common_parents(self):
+        error = ItalianRestaurantManyParentsAbstract.check()[0]
+        clash = Restaurant._meta.get_field("place_ptr")
+        field = ItalianRestaurantManyParentsAbstract._meta.get_field("place_ptr")
+        expected_error = Error(
+            "The field '%s' clashes with the field '%s' "
+            "from model '%s'." % (field.name, clash.name, clash.model._meta),
+            obj=field,
+            id="models.E006",
+        )
+        self.assertEqual(error.level, expected_error.level)
+        self.assertEqual(error.msg, expected_error.msg)
+        self.assertEqual(error.hint, expected_error.hint)
+        self.assertEqual(error.obj.name, expected_error.obj.name)
```

Comment:
This should work without checking each attribute separately, e.g.
```python
        class ItalianRestaurantManyParentsAbstract(ItalianRestaurant, Place):
            pass

        field = ItalianRestaurantManyParentsAbstract._meta.get_field("place_ptr")
        self.assertEqual(
            ItalianRestaurantManyParentsAbstract.check(),
            [ 
                Error(
                    "The field 'place_ptr' clashes with the field 'place_ptr' from "
                    "model 'model_inheritance.restaurant'.",
                    obj=field,
                    id="models.E006",
                )
            ],
        )
```

Reply:
Updated the test.